File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int min=prices[0], max=0;
        
        for(int i=1; i<prices.length; i++) {
            if (min > prices[i]) {
                min = prices[i];
            }
            else {
                max = prices[i] - min > max ? prices[i]-min : max;
            }
        }
        return max;
        
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int maxProfit = 0;
        int currentMin = prices[0];
        
        for(int i=1; i<prices.size(); i++) {
            maxProfit = max(maxProfit, prices[i] - currentMin);
            currentMin = min(currentMin, prices[i]);
        }
        
        return maxProfit;
    }
};